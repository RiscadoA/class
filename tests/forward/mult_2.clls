proc main() {
    cut {
        println("recv x");
        recv a(x);
        println("wait a");
        wait a;
        println("close x");
        close x
        |a:send wait; close|
        println("send y");
        send a(y. println("wait y"); wait y; println("end"); ());
        println("forward");
        fwd a b
        |b:close|
        println("close b");
        close b
    }
};;
