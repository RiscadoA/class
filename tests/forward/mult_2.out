close b
send y
forward
recv x
wait a
close x
wait y
end
